### PR TITLE
The rename games part 1 of many

### PR DESCRIPTION
**Pull request by davidfowl**
- React to rename of EndPoint to ConnectionHandler
- Rename UseSockets to UseConnections
- Rename MapEndPoint to MapConnectionHandler
- Rename HttpSocketOptions to HttpConnectionOptions

See https://github.com/aspnet/KestrelHttpServer/pull/2413

PS: I'd like to merge this as soon as possible since it's a large change that affects many many files. I have more follow up renames to do as well.

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Tests.Utils/TestClient.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ public TestClient(bool synchronousCallbacks = false, IHubProtocol protocol = nul
         }
 
         public async Task<Task> ConnectAsync(
-            EndPoint endPoint,
+            Connections.ConnectionHandler handler,
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Tests.Utils/TestClient.cs]: I'm going to rename DefaultConnectionContext in a follow up PR

File: test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs [hunk 2]
```diff
@@ -75,7 +75,7 @@ public HttpConnectionDispatcherTests(ITestOutputHelper output) : base(output)
                 context.Request.Path = "/foo";
                 context.Request.Method = "POST";
                 context.Response.Body = ms;
-                var httpSocketOptions = new HttpSocketOptions { TransportMaxBufferSize = 4, ApplicationMaxBufferSize = 4 };
+                var httpSocketOptions = new HttpConnectionOptions { TransportMaxBufferSize = 4, ApplicationMaxBufferSize = 4 };
```
Comment (JamesNK) [inline test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs]: boop

Comment (davidfowl) [reply test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs]: 😝

**Comment by davidfowl** [comment]
> It looks ok but do a text search for endpoint, etc, so see if there are rogue variables and parameters that haven't been renamed.

The only things I see in VS code are some test names, comments and random type script things.

**Comment by davidfowl** [comment]
```
not ok 74 hubConnection using messagepack over LongPolling transport can be restarted
    ---
      - message: Expected Error: Server returned handshake error: Handshake was canceled. to be undefined.
        operator: toBeUndefined
        expected: 
        actual: Error: Server returned handshake error: Handshake was canceled.
```

**Comment by davidfowl** [comment]
not ok 74 hubConnection using messagepack over LongPolling transport can be restarted

**Pull request merged**